torchaudio/prototype/models/hifi_gan.py

```diff
@@ -0,0 +1,307 @@
+                    padding=get_padding(kernel_size, dilation[2]),
+                ),
+            ]
+        )
+
+        self.convs2 = nn.ModuleList(
+            [
+                Conv1d(channels, channels, kernel_size, 1, dilation=1, padding=get_padding(kernel_size, 1)),
+                Conv1d(channels, channels, kernel_size, 1, dilation=1, padding=get_padding(kernel_size, 1)),
+                Conv1d(channels, channels, kernel_size, 1, dilation=1, padding=get_padding(kernel_size, 1)),
+            ]
+        )
+        self.lrelu_slope = lrelu_slope
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Args:
+            x (Tensor): input of shape ``(batch_size, channels, time_length)``.
+        Returns:
+            Tensor of the same shape as input.
+        """
+        for conv1, conv2 in zip(self.convs1, self.convs2):
+            xt = F.leaky_relu(x, self.lrelu_slope)
+            xt = conv1(xt)
+            xt = F.leaky_relu(xt, self.lrelu_slope)
+            xt = conv2(xt)
+            x = xt + x
+        return x
+
+
+class ResBlock2(torch.nn.Module):
+    """Residual block of type 2 for HiFiGAN Generator :cite:`NEURIPS2020_c5d73680`.
+    Args:
+        channels (int): Number of channels in the input features.
+        kernel_size (int, optional): Kernel size for 1D convolutions. (Default: ``3``)
+        dilation (tuple of 2 ints, optional): Dilations for each 1D convolution. (Default: ``(1, 3)``)
+        lrelu_slope (float): Slope of leaky ReLUs in activations.
+    """
+
+    def __init__(
+        self, channels: int, kernel_size: int = 3, dilation: Tuple[int, int] = (1, 3), lrelu_slope: float = 0.1
+    ):
+        super(ResBlock2, self).__init__()
+        self.convs = nn.ModuleList(
+            [
+                Conv1d(
+                    channels,
+                    channels,
+                    kernel_size,
+                    1,
+                    dilation=dilation[0],
+                    padding=get_padding(kernel_size, dilation[0]),
+                ),
+                Conv1d(
+                    channels,
+                    channels,
+                    kernel_size,
+                    1,
+                    dilation=dilation[1],
+                    padding=get_padding(kernel_size, dilation[1]),
+                ),
+            ]
+        )
+        self.lrelu_slope = lrelu_slope
+
+    def forward(self, x: torch.Tensor):
+        """
+        Args:
+            x (Tensor): input of shape ``(batch_size, channels, time_length)``.
+        Returns:
+            Tensor of the same shape as input.
+        """
+        for c in self.convs:
+            xt = F.leaky_relu(x, self.lrelu_slope)
+            xt = c(xt)
+            x = xt + x
+        return x
+
+
+def get_padding(kernel_size, dilation=1):
+    """Find padding for which 1D convolution preserves the input shape."""
+    return int((kernel_size * dilation - dilation) / 2)
+
+
+def hifigan_model(
```

Comment:
Considering we may support training HiFiGAN in the future, we can call the factory function `hifigan_generator` in this PR. And we can add `hifigan_discriminator` and `hifigan_model` (which comprises `hifigan_generator` and `hifigan_discriminator`).
```suggestion
def hifigan_generator(
```

Reply:
Done